kvdb-rocksdb/src/lib.rs

```diff
@@ -460,7 +472,21 @@ impl Database {
 		}
 		self.stats.tally_bytes_written(stats_total_bytes as u64);
 
-		cfs.db.write_opt(batch, &self.write_opts).map_err(other_io_err)
+		let res = cfs.db.write_opt(batch, &self.write_opts).map_err(other_io_err)?;
+
+		// If we have written more data than what we want to have stored in a `sst` file, we force compaction.
+		// We also ensure that we only compact once per minute.
+		//
+		// Otherwise, rocksdb read performance is going down, after e.g. a warp sync.
+		if stats_total_bytes > self.config.compaction.initial_file_size as usize &&
+			self.last_compaction.lock().elapsed() > Duration::from_secs(60)
```

Comment:
Any ideas how long compaction lasts? 60s sounds too often - but it is a pure guess from my side.